recipes/lzma_sdk/9.20/conanfile.py

```diff
@@ -0,0 +1,141 @@
+import os
+from conans import ConanFile, tools, AutoToolsBuildEnvironment, VisualStudioBuildEnvironment
+
+
+""" This older lzma release is used to build 7zip (to extract its sources).
+    That's why this lzma_sdk/9.20 conan package includes a 7zDec executable.
+
+    Recent versions of 7zip are distributed as 7z archives.
+    So we need 7zip itself to uncompress its sources.
+    To break this cycle, we have 2 options:
+    * download an executable and run it
+    * bootstrap using a previous version (lzma_sdk/9.20) which sources are in .tar.bz2.
+    Right now it would be a loop in the Conan graph (if we'ld use the same recipe name)
+"""
+
+
+class PackageLzmaSdk(ConanFile):
+    name = "lzma_sdk"
+    version = "9.20"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "LZMA provides a high compression ratio and fast decompression, so it is very suitable for embedded applications."
+    license = ("LZMA-exception",)
+    author = "Conan Community"
+    homepage = "https://www.7-zip.org/sdk.html"
+    topics = ("conan", "lzma", "zip", "compression", "decompression")
+    settings = "os_build", "arch_build", "compiler"
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and os.environ.get("CONAN_BASH_PATH", None) is None:
+            self.build_requires("msys2/20161025")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.unlink(os.path.join(self.source_folder, "7zr.exe"))
+        os.unlink(os.path.join(self.source_folder, "lzma.exe"))
+
+    @property
+    def _msvc_build_dirs(self):
+        return (
+            [os.path.join("C","Util","7z"), [["7zDec.exe"], []]],
+            [os.path.join("C","Util","Lzma"), [["LZMAc.exe"], []]],
+            # [os.path.join("C","Util","LzmaLib"), [["LZMA.dll"], ["LZMA.lib"]]],
+            # [os.path.join("C","Util","SfxSetup"), [["7zS2.sfx"], []]],
+            [os.path.join("CPP","7zip","UI","Console"), [["7z.exe"], []]],
+            [os.path.join("CPP","7zip","Bundles","Alone7z"), [["7zr.exe"], []]],
+            # [os.path.join("CPP","7zip","Bundles","Format7zExtractR"), [["7zxr.dll"], ["7zxr.lib"]]],
+            # [os.path.join("CPP","7zip","Bundles","Format7zR"), [["7zra.dll"], ["7zra.lib"]]],
+            [os.path.join("CPP","7zip","Bundles","LzmaCon"), [["lzma.exe"], []]],
+        )
+
+    @property
+    def _msvc_cpu(self):
+        return {
+            'x86_64': 'AMD64',
+            'x86': 'x86',
+        }[str(self.settings.arch_build)]
+
+    @property
+    def _autotools_build_dirs(self):
+        es = ".exe" if self.settings.os_build == "Windows" else ""
+        return (
+            [os.path.join("C","Util","7z"), [["7zDec{}".format(es)], []]],
+            # [os.path.join("C","Util","Lzma"), [["lzma{}".format(es)], []]],
```

Comment:
remove not needed line?

Reply:
The commented out libs create libs which may possibly be useful for some users.
I tried to package some of them and write tests, but did not succeed.